engine/wait_for_process_completion_channel.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023 xCherryIO organization
+
+// Copyright (c) 2023 xCherryIO Organization
+// SPDX-License-Identifier: Apache-2.0
+
+package engine
+
+import (
+	"fmt"
+	"github.com/xcherryio/xcherry/common/log"
+	"time"
+)
+
+type WaitForProcessCompletionChannelsPerShardImpl struct {
+	shardId int32
+	logger  log.Logger
+
+	processor ImmediateTaskProcessor
+
+	// processExecutionId : channel
+	channelMap map[string]chan string
```

Comment:
similarly here, operating a map under concurrency needs some locking to protect it